src/components/Button/index.tsx

```diff
@@ -1,9 +1,39 @@
 'use client';
-import { ButtonHTMLAttributes, DetailedHTMLProps } from 'react';
+import { ButtonHTMLAttributes, DetailedHTMLProps, ReactNode, useState } from 'react';
 import styles from './button.module.css';
+import { FontAwesomeIcon, FontAwesomeIconProps } from '@fortawesome/react-fontawesome';
+import { IconDefinition, library, icon, IconProp } from "@fortawesome/fontawesome-svg-core";
+
+type ButtonProps = {
+  icon?: IconDefinition;
+  red?: boolean;
+  children?: ReactNode;
+  className?: string;
+} & DetailedHTMLProps<
+  ButtonHTMLAttributes<HTMLButtonElement>, 
+  HTMLButtonElement>;
+
 export default function Button({
+  red,
+  className,
   ...props
-}: DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>) {
+}: ButtonProps) {
+  const [isPending, setIsPending] = useState(false); 
+  
+  const setPending = (bool_:boolean) => {
+    setIsPending(bool_);
+  };
+
+  // If icon exists add to libary
+  if(props.icon)
+		library.add(props.icon);
 
-  return <button {...props} className={props.className + ' ' + styles.button} />
+  return (
+  <button {...props} className={`${className} ${red? styles.primaryButton : styles.button} disabled:opacity-25 flex align-center justify-items-center`}>
```

Comment:
this could possibly be changed to `props.className.includes('red')` and the red boolean flag can be removed and passed down through the className - We could also add a variant map for a range of colors 

Ex:
```ts

export default function Button({
  ...props
}: DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
> & {
  icon?: IconDefinition;
  variant?:
    | 'Primary'
    | 'Secondary'
    | 'Warning'
    | 'Info'
    | 'Danger'
    | 'Success'
    | 'Neutral';
}) {
  const className = [props.className, props.variant].join(' ');

  return (
    <button {...{ ...props, className }}>
      {props.icon && <Icon className="inherit" icon={props.icon} />}
      {!props.icon && props.children}
    </button>
  );
}
```

```css


svg.Primary {
  color: var(--green);
}
svg.Primary:not(:disabled):hover {
  color: var(--green-hover);
}

svg.Secondary {
  color: var(--red);
}
svg.Secondary:not(:disabled):hover {
  color: var(--red-hover);
}

/* ... */ 
```

Reply:
I've introduced the suggested changes as described by passing through className